lib/pbench/server/hierarchy.py

```diff
@@ -0,0 +1,444 @@
+        """Output Messages for each list of controllers, files,
+        directories, etc depicting their cause of being added to
+        that list or dictionary
+        """
+        lead_asterisk = "\t* " if hierarchy == "archive" else "\t"
+        if controller in self.validation_list[key]._dict:
+            if key not in ["subdir_status_indicators", "prefix_status_indicators"]:
+                fp.write(f"{lead_asterisk}{self.validation_list[key]._get_msg()}\n")
+                if hierarchy == "archive":
+                    cnt = self.output_format(
+                        fp, self.validation_list[key]._dict[controller], cnt, "archive"
+                    )
+                else:
+                    cnt = self.output_format(
+                        fp, self.validation_list[key]._dict[controller], cnt
+                    )
+            elif "subdirs" in self.validation_list[key]._dict[controller]:
+                fp.write(f"{lead_asterisk}{self.validation_list[key]._get_msg()}\n")
+                cnt += 1
+            elif "prefix_dir" in self.validation_list[key]._dict[controller]:
+                fp.write(f"{lead_asterisk}{self.validation_list[key]._get_msg()}\n")
+                cnt += 1
+        return cnt
+
+    def check_tab_format(self, hierarchy):
+        """Check the format of tab acc. to hierarchy"""
+
+        lead_tab = ""
+        if hierarchy == "controller":
+            lead_tab = "\t"
+        elif hierarchy == "archive":
+            lead_tab = "\t  "
+
+        return lead_tab
+
+    def output_format(self, fp, controller_list, cnt, hierarchy=False):
+        """Output format of list"""
+
+        lead_tab = self.check_tab_format(hierarchy) if hierarchy else "\t\t"
+        if hierarchy == "archive":
+            fp.write("\t  ++++++++++\n")
+        self.output_list(fp, lead_tab, controller_list)
+        if hierarchy == "archive":
+            fp.write("\t  ----------\n")
+        cnt += 1
+        return cnt
+
+    def output_list(self, fp, lead_tab, controller_list):
+        """ Writing list value to file"""
+        for value in sorted(controller_list):
+            fp.write(f"{lead_tab}{value}\n")
+        return
+
+
+class DictOfList(object):
+    """Class used to create and add elements to Dictionary"""
+
+    def __init__(self, msg):
+        self._dict = OrderedDict()
+        self._msg = msg
+
+    def _get_msg(self):
+        return self._msg
+
+    def __setitem__(self, key, value):
+        if key in self._dict:
+            self._dict[key].append(value)
+        else:
+            self._dict[key] = [value]
+
+
+class List(object):
+    """Class used to create and add elements to Dictionary"""
+
+    def __init__(self, msg):
+        self._list = list()
+        self._msg = msg
+
+    def _get_msg(self):
+        return self._msg
+
+    def additem(self, value):
+        self._list.append(value)
+
+    def getlist(self):
+        return self._list
```

Comment:
Use the standard dunder operators to simply make this "annotated list" behave like a normal Python list.

Reply:
sure